runtime/router/trie_with_speical_cases.go

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package router
+
+import "net/http"
+
+// set sets the handler for given path, creates new child node if necessary
+// lastKeyCharSlash tracks whether the previous key char is a '/', used to decide it is a pattern or not
+// when the current key char is ':'. lastPathCharSlash tracks whether the previous path char is a '/',
+// used to decide it is a pattern or not when the current path char is ':'.
+func (t *tnode) setForWhitelistedPath(path string, value http.Handler, lastKeyCharSlash, lastPathCharSlash bool) error {
```

Comment:
rather than duplicating the entire code for whitelist path, can we just add it in trie.go. It will make it a single place and easier to maintain.

Reply:
I have purposefully forked the logic for whitelisted paths because this will enable us to support adding more special cases for whitelisted paths in future. That way the original remains untoched.

Reply:
I agree with Abhishek here, this is going to be a pain to keep the 2 versions in sync.  Imagine having to fix a bug in the get/set methods, it is possible to miss the fixes in one or the other place. Also, reviewing is a pain, I have to check which of the logic is new here.

Reply:
Right now itself the logic is hard to understand. The code will become more messy if we combine this logic. I will leave it here and let you guys take a call on this.

Reply:
resolving as duplication of the entire whitelist code is removed.